aws_lambda_powertools/utilities/data_classes/dynamo_db_stream_event.py

```diff
@@ -4,12 +4,30 @@
 from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
 
 
+class AttributeValueType(Enum):
+    Binary = "B"
+    BinarySet = "BS"
+    Boolean = "BOOL"
+    List = "L"
+    Map = "M"
+    Number = "N"
+    NumberSet = "NS"
+    Null = "NULL"
+    String = "S"
+    StringSet = "SS"
+
+
 class AttributeValue(DictWrapper):
     """Represents the data for an attribute
 
     Documentation: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_AttributeValue.html
     """
 
+    @property
+    def get_type(self) -> AttributeValueType:
```

Comment:
What about a generic “get_value” method that can inspect the type and return the proper value?

Reply:
@jplock -  But then i would need to type hint as a Union? Coz it is kind of nice having the type hinting

Reply:
That’s true. I end up using the boto3.dynamodb.types TypeDeserializer directly anyway

Reply:
I happen to need the **get_type**, but a **get_value** is an obvious idea, should have also suggested that.

Reply:
Thanks @jplock for pointing me to https://boto3.amazonaws.com/v1/documentation/api/latest/_modules/boto3/dynamodb/types.html

I am thinking of a type hinted version of this.

Reply:
@jplock @gwlester @heitorlessa 

I added "value" implementation, as it is not quite a deserialize. But maybe it should be and do the binary handling to? However that would be a breaking change.

Reply:
I'm fine either way. 'get_value' would be more consistent and inline with the operation it does here than 'value'

Reply:
@heitorlessa renamed